udata/core/reuse/models.py

```diff
@@ -105,6 +105,8 @@ def post_save(cls, sender, document, **kwargs):
             cls.on_create.send(document)
         else:
             cls.on_update.send(document)
+        if document.deleted:
```

Comment:
Do we want to send the `on_update` just above? To me we should check at the start if it has been deleted and not perform other updates in that case.